src/rez/pip.py

```diff
@@ -267,16 +268,30 @@ def pip_install_package(source_name, pip_version=None, python_version=None,
         tools = []
         src_dst_lut = {}
 
-        for installed_file in distribution.list_installed_files(allow_fail=True):
-            source_file = os.path.normpath(os.path.join(destpath, installed_file[0]))
+        for installed_file in distribution.list_installed_files():
+            # a relative directory that needs to be appended without the
+            # os.sep (via os.path.join) in order for normpath to return the
+            # correct path, this is in order to match the wheel RECORD file
+
+            # Example
+
+            # destpath = /tmp/pip-HBkRGa-rez/re]z_staging/python
+            # installed_file[0] = ../../bin/pydocstyle
+            # normpath of /tmp/pip-HBkRGa-rez/rez_staging/python../../bin/pydocstyle =  /tmp/pip-HBkRGa-rez/rez_staging/bin/pydocstyle OK
+            # normapth of /tmp/pip-HBkRGa-rez/rez_staging/python/../../bin/pydocstyle = /tmp/pip-HBkRGa-rez/bin/pydocstyle WRONG
+            if installed_file[0].startswith("."):
```

Comment:
I would make this clearer, perhaps like so:

```
installed_filepath = installed_file[0]
bin_prefix = os.sep.join('..', '..', 'bin') + os.sep
...

if installed_filepath.startswith(bin_prefix):
    # account for extra parentdir as explained above
    installed = os.path.join(destpath, '_', installed_filepath)
else:
    ...
```

Reply:
Never knew that normpath will normalize a pathname by collapsing **all** redundant separators and up-level references. I always thought it was limited to dots and slashes, awesome. :+1: